src/overlay/Peer.cpp

```diff
@@ -54,14 +54,17 @@ using namespace soci;
 namespace
 {
 // Check the signature(s) in `tx`, adding the result to the signature cache in
-// the process. For performance reasons, this function should only be called
-// from the overlay thread.
+// the process. This function does nothing if background signature verification
+// is disabled and/or the current thread is not the overlay thread.
 void
 checkTransactionSignature(AppConnector& app, TransactionFrameBaseConstPtr tx)
```

Comment:
Given that this function can no-op, maybe a more descriptive name would be "populateSignatureCache". I also am not a huge fan of dictating logic based on the current thread (beyond asserts) and it seems like a footgun to rely on thread PIDs. Is it possible to keep this check as an assert and move the conditional logic to the caller?

Reply:
I renamed this to `populateSignatureCheck` and switched the check back to an assert. The check now occurs in the `CapacityTrackedMessage` constructor and relies on the `useBackgroundThread` function rather than checking thread ids.